### PR TITLE
Revert Dotnet 5 SDK for core-tools

### PR DESCRIPTION
**Pull request by CooperLink**
### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Comment by balag0** [comment]
We are reverting because this is not needed or something is failing?

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-core-tools.Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
 #-------------------------------------------------------------------------------------------------------------
 
-FROM mcr.microsoft.com/dotnet/sdk:5.0
+FROM mcr.microsoft.com/dotnet/sdk:3.1
```
Comment (pragnagopa) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-core-tools.Dockerfile]: Please see https://hub.docker.com/_/microsoft-dotnet-sdk for the right tag

**Review by pragnagopa — CHANGES_REQUESTED** [review]
fix image tag

**Comment by pragnagopa** [comment]
> We are reverting because this is not needed or something is failing?

Yes, .Net 5 SDK is not needed as host is not building in Core-tools image

**Comment by pragnagopa** [comment]
@CooperLink - can you please paste the errors for building this image if using .Net 5 SDK?

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-core-tools.Dockerfile (hunk 1 above)
Comment (CooperLink) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-core-tools.Dockerfile]: Updated in most recent commit to use dotnet/core/sdk:3.1 as it did before PR #515

**Comment by CooperLink** [comment]
@pragnagopa  afaik there are no errors on build for this image. The issue is that we tag the image dotnet3-core-tools and the devcontainers team relies on this image building with 3.1.  

To maintain consistency and clarity for our docker images, I think that we should continue to provide the dotnet3-core-tools and possibly the powershell7-core-tools images that do not build host with dotnet3.1 sdk

See issue
https://github.com/Azure/azure-functions-docker/issues/518

**Comment by pragnagopa** [comment]
> To maintain consistency and clarity for our docker images, I think that we should continue to provide the dotnet3-core-tools and possibly the powershell7-core-tools images that do not build host with dotnet3.1 sdk

Thanks! Lets go ahead and move them back to .Net Core 3.1 SDK in docker files where we do not build host

**Review by pragnagopa — APPROVED** [review]
👍

**Pull request merged**